ui/src/app/lib/atlasmap-data-mapper/components/document.definition.component.ts

```diff
@@ -53,8 +53,8 @@ import { ModalWindowComponent } from './modal.window.component';
                 </div>
                 <div [attr.class]="searchMode ? 'fieldListSearchOpen' : 'fieldList'" style="overflow:auto;"
                     (scroll)="handleScroll($event)">
-                    <div *ngFor="let docDef of cfg.getDocs(isSource)" #docDetail class="docIdentifier" [attr.id]='docDef.name'>
-                        <div class="card-pf-title documentHeader" tooltip="{{ docDef.fullyQualifiedName }}" placement="bottom"
+                    <div *ngFor="let docDef of cfg.getDocs(isSource)" #docDetail class="docIdentifier" [attr.id]='docDef.shortName'>
```

Comment:
The Angular change detector will evaluate this line multiple times in a second, thus executing `cfg.getDocs(isSource)` on each pass. I do not know the internals of that function but I'd move it to a `docDefs` property in the component class and populate it at `ngOnInit()`. Otherwise you'll wind up running this method every few milliseconds.

Reply:
Thanks for the review!
There are 2 types of docs here, and we are retrieving sourceDocs if isSource=true, otherwise targetDocs. Do we want to add 2 `*ngIf` here to add this `<div>` conditionally?

Reply:
If `isSource === false`, that <div> will not even show up so the conditional would become redundant.

Reply:
I think cfg.getDocs(false) returns target documents, am I missing something?
https://github.com/atlasmap/atlasmap/blob/master/ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts#L174-L181
btw these method stink...

Reply:
Ah never mind, `isSource` is never changed once initialized, so `ngOnInit()` can decide which is required. Same for `Sources`/`Targets` label.